src/ghdldrv/ghdldrv.adb

```diff
@@ -1660,7 +1663,46 @@ package body Ghdldrv is
                              Args : Argument_List)
    is
       pragma Unreferenced (Cmd);
+   begin
+      Gen_Makefile(Args, False);
+   end Perform_Action;
+
+   --  Command Gen_Depends.
+   type Command_Gen_Depends is new Command_Comp with null record;
+   function Decode_Command (Cmd : Command_Gen_Depends; Name : String)
+                           return Boolean;
+   function Get_Short_Help (Cmd : Command_Gen_Depends) return String;
+   procedure Perform_Action (Cmd : in out Command_Gen_Depends;
+                             Args : Argument_List);
+
+   function Decode_Command (Cmd : Command_Gen_Depends; Name : String)
+                           return Boolean
+   is
+      pragma Unreferenced (Cmd);
+   begin
+      return Name = "--gen-depends";
+   end Decode_Command;
 
+   function Get_Short_Help (Cmd : Command_Gen_Depends) return String
+   is
+      pragma Unreferenced (Cmd);
+   begin
+      return "--gen-depends [OPTS] UNIT [ARCH]"
+        & "Generate dependencies of UNIT";
+   end Get_Short_Help;
+
+   procedure Perform_Action (Cmd : in out Command_Gen_Depends;
+                             Args : Argument_List)
+   is
+      pragma Unreferenced (Cmd);
+   begin
+      Gen_Makefile(Args, True);
+   end Perform_Action;
+
+   procedure Gen_Makefile (Args : Argument_List; Only_Depends : Boolean)
+      -- generate a makefile on stdout
+      -- for --gen-depends (Only_Depends) rules and phony targets are omittted
```

Comment:
Usually the comments appear before the function.  You can even move them just before the specification (aka the prototype).